pkg/scaling/scale_handler.go

```diff
@@ -288,7 +285,82 @@ func (h *ScaleHandler) parseAuthRef(triggerAuthRef *kedav1alpha1.ScaledObjectAut
+			// resource doesn't exist
+			h.logger.Error(err, "Target resource doesn't exist", "resource", obj.Status.ScaleTargetGVKR.GVKString(), "name", obj.Spec.ScaleTargetRef.Name)
+			return nil, "", err
+		}
+
+		withPods := &duckv1.WithPod{}
+		if err := duck.FromUnstructured(unstruct, withPods); err != nil {
+			h.logger.Error(err, "Cannot convert unstructured into PodSpecable Duck-type", "object", unstruct)
+		}
+
+		if withPods.Spec.Template.Spec.Containers == nil {
+			h.logger.Info("There aren't any containers in the ScaleTarget", "resource", obj.Status.ScaleTargetGVKR.GVKString(), "name", obj.Spec.ScaleTargetRef.Name)
+			return nil, "", fmt.Errorf("No containers found")
+		}
+
+		return withPods, obj.Spec.ScaleTargetRef.ContainerName, nil
+	}
+
+	// TODO: implement this for ScaledJobs!! 
+	return nil, "", fmt.Errorf("resolvePods is only implemented for ScaledObjects so far")
+}
+
+// GetScaledObjectScalers returns list of Scalers for the specified ScaledObject
+func (h *scaleHandler) getScalers(scalableType *kedav1alpha1.WithTriggers, withPods *duckv1.WithPod, containerName string) ([]scalers.Scaler, error) {
+	scalersRes := []scalers.Scaler{}
+
+	resolvedEnv, err := h.resolveContainerEnv(&withPods.Spec.Template.Spec, containerName, scalableType.Namespace)
+	if err != nil {
+		return scalersRes, fmt.Errorf("error resolving secrets for ScaleTarget: %s", err)
+	}
+
+	for i, trigger := range scalableType.Spec.Triggers {
+		authParams, podIdentity := h.parseScaledObjectAuthRef(trigger.AuthenticationRef, scalableType, &withPods.Spec.Template.Spec)
+
+		if podIdentity == kedav1alpha1.PodIdentityProviderAwsEKS {
+			serviceAccountName := withPods.Spec.Template.Spec.ServiceAccountName
+			serviceAccount := &corev1.ServiceAccount{}
+			err = h.client.Get(context.TODO(), types.NamespacedName{Name: serviceAccountName, Namespace: scalableType.Namespace}, serviceAccount)
+			if err != nil {
+				closeScalers(scalersRes)
+				return []scalers.Scaler{}, fmt.Errorf("error getting service account: %s", err)
+			}
+			authParams["awsRoleArn"] = serviceAccount.Annotations[kedav1alpha1.PodIdentityAnnotationEKS]
+		} else if podIdentity == kedav1alpha1.PodIdentityProviderAwsKiam {
+			authParams["awsRoleArn"] = withPods.Spec.Template.ObjectMeta.Annotations[kedav1alpha1.PodIdentityAnnotationKiam]
+		}
+
+		scaler, err := h.getScaler(scalableType.Name, scalableType.Namespace, trigger.Type, resolvedEnv, trigger.Metadata, authParams, podIdentity)
+		if err != nil {
+			closeScalers(scalersRes)
+			return []scalers.Scaler{}, fmt.Errorf("error getting scaler for trigger #%d: %s", i, err)
+		}
+
+		scalersRes = append(scalersRes, scaler)
+	}
+
+	return scalersRes, nil
+}
+
+func (h *scaleHandler) parseScaledObjectAuthRef(triggerAuthRef *kedav1alpha1.ScaledObjectAuthRef, scalableType *kedav1alpha1.WithTriggers, podSpec *corev1.PodSpec) (map[string]string, string) {
```

Comment:
we should rename this func, since it is not `ScaledObject` exclusive and change the name of `kedav1alpha1.ScaledObjectAuthRef` too